### PR TITLE
[Provisioner] Fix Azure disk tier explicitly shown in resources str

### PR DESCRIPTION
**Pull request by cblmemo**
To reproduce:

```bash
$ sky launch --cloud azure
Clusters
NAME                           LAUNCHED        RESOURCES                                                 STATUS   AUTOSTOP  COMMAND                       
sky-ccca-txia                  a few secs ago  1x Azure(Standard_D8s_v5, disk_tier=medium)               INIT     -         sky launch --cloud azure
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
    - [x] Previous reproducible code
    - [x] `sky launch -t Standard_D8_v5 --disk-tier medium` and correctly throws an error
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
    - [x] tests/test_smoke.py::test_azure_disk_tier
    - [x] tests/test_smoke.py::test_azure_best_tier_failover
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/azure.py [hunk 1]
```diff
@@ -295,6 +295,20 @@ def make_deploy_resources_variables(
...
+            all_tiers = list(reversed(resources_utils.DiskTier))
+            start_index = all_tiers.index(
+                Azure._translate_disk_tier(r.disk_tier))
+            while start_index < len(all_tiers):
+                disk_tier = all_tiers[start_index]
+                ok, _ = Azure.check_disk_tier(r.instance_type, disk_tier)
+                if ok:
+                    return disk_tier
+                start_index += 1
+            return None
```
Comment (Michaelvll) [inline sky/clouds/azure.py:310]: Does this mean a user specifying `--disk-tier medium` may get an instance with disk_tier low?

**Review by Michaelvll — COMMENTED** [review]
Thanks for submitting the PR @cblmemo! Left a comment.

File: sky/clouds/azure.py (hunk 1 above)
Comment (cblmemo) [reply sky/clouds/azure.py]: Good point! We should only do failover if the disk tier is None or BEST. Just pushed a commit to fix that.

Comment (cblmemo) [reply sky/clouds/azure.py]: Just re-run every tests and passed

File: sky/clouds/azure.py [hunk 2]
```diff
@@ -303,75 +320,38 @@ def make_deploy_resources_variables(
...
+            ok, _ = Azure.check_disk_tier(resources.instance_type,
+                                          resources.disk_tier)
             if not ok:
                 return ([], [])
             # Treat Resources(Azure, Standard_NC4as_T4_v3, T4) as
             # Resources(Azure, Standard_NC4as_T4_v3).
             resources = resources.copy(
                 accelerators=None,
-                disk_tier=disk_tier,
+                disk_tier=resources.disk_tier,
```
Comment (Michaelvll) [inline sky/clouds/azure.py:340]: If we are using the same `disk_tier`, we don't have to pass the argument.
```suggestion
```

File: sky/clouds/azure.py [hunk 3]
```diff
@@ -303,75 +320,38 @@ def make_deploy_resources_variables(
...
-                                                   resources.disk_tier)
+                ok, _ = Azure.check_disk_tier(instance_type,
+                                              resources.disk_tier)
                 if not ok:
                     continue
                 r = resources.copy(
                     cloud=Azure(),
                     instance_type=instance_type,
-                    disk_tier=disk_tier,
+                    disk_tier=resources.disk_tier,
```
Comment (Michaelvll) [inline sky/clouds/azure.py:354]: Similarly, we should not pass the `disk_tier` if not changed. : )
```suggestion
```

**Review by Michaelvll — APPROVED** [review]
Thanks for the fix @cblmemo! LGTM.

**Pull request merged**